### DISCUSSION TITLE
What is the effect of batch_group_size

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#2064] [General Q&A]
Hi,

Previously I ran into Out of Memory errors  so I tried to decrease batch_group_size to 0.

Now that I have found a bunch of working hyperparameters that work I am wondering what the effects are of having disabled batch_group_size.

I don't understand the source code that tells that batch_group_size sets the randomization of samples within the mini batch
> Range of batch randomization after sorting
                sequences by length. It shuffles each batch with bucketing to gather similar lenght sequences in a
                batch. Set 0 to disable. Defaults to 0.

Will it improve the training speed / model accuracy ? If I double the batch size from 32 to 64 should I also double batch group size ? Is it a good / bad idea to disable it ? 

**EDIT (after some basic "quick" benchmarking) : I tried during fine tuning of a model to vary `batch_group_size` and measure the time to reach step 300. I noticed that for every run the training with `batch_group_size = 5` (11min30s) was roughly 25% faster than with `batch_group_size=0` (13min21s). Regarding the VRAM consumption the value I measured was also roughly 25% smaller with 5 than with 0. So pretty interesting. The fine tuning was done with 0.5k samples.**

Thanks in advance for the clarifications

**Comment by p0p4k** [comment]
Take a guess what is happening here.
https://github.com/coqui-ai/TTS/blob/d6ad9a05b40e9f3aa876ecfa1c0d23339f373df4/TTS/tts/datasets/dataset.py#L346-L350
https://github.com/coqui-ai/TTS/blob/d6ad9a05b40e9f3aa876ecfa1c0d23339f373df4/TTS/tts/datasets/dataset.py#L301-L309
![image](https://user-images.githubusercontent.com/8834712/194707083-090e9010-ed91-4987-baa1-84ffe26ea100.png)
Good luck!

**Reply by Ca-ressemble-a-du-fake** [reply]
I don't get it :laughing: ! Is it obvious ? I just see that it shuffles samples but I don't understand the point of this. Anyway thanks for your reply!
